lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -413,9 +413,11 @@ struct ElementwiseInlineAsmOpConversion
     // [return_value][op.getPackedElement()].
     SmallVector<SmallVector<Value>> ret(op->getNumResults());
     for (int i = 0; i < op->getNumResults(); i++) {
+      int structIdx = 0;
       for (int j = 0; j < op.getPackedElement(); j++) {
         auto val = asmRetTypes.size() > 1
-                       ? extract_val(asmResults, i * op.getPackedElement() + j)
+                       ? extract_val(asmResults,
+                                     i * op.getPackedElement() + structIdx++)
```

Comment:
up to you but I think this might be easier to see if it's written as an actual `if` statement.  I had to think about whether a ternary evaluates the side-effects of structIdx++.

Reply:
replaced